pymodbus/client/base.py

```diff
@@ -60,7 +60,6 @@ def __init__(
         **kwargs: Any,
     ) -> None:
         """Initialize a client instance."""
-        ModbusClientMixin.__init__(self)
```

Comment:
This is no good, we use the init method to inject special tests. Is there a special reason why mixin no longer can have a init() ?

Reply:
"self" is the wrong type for that init, now. the alternative would be type: ignore so i'll have to switch to that.

Reply:
"self" is of course a wrong type, it should be "ModbusBaseClient | ModbusAsyncBaseClient",